File: python/5073.py
def main(a, b, c) :
    list_triangle = [a, b, c]
    list_triangle.sort()
    if(list_triangle[0]+list_triangle[1] <= list_triangle[2]):
        return "Invalid"
    if (a==b==c):
        return "Equilateral"
    elif (a!=b and a!=c and b!=c) :
        return "Scalene"
    else :
        return "Isosceles"

while True:
    a, b, c = list(map(int, input().split()))
    if(a == b == c == 0) :
        break
    print(main(a, b, c))

File: algorithm/src/5000~10000/5073.kt
package `5000~10000`

import java.util.Scanner

fun main() = with(Scanner(System.`in`)) {
    while (true) {
        val triangle : Array<Int> = arrayOf(nextInt(), nextInt(), nextInt())
        triangle.sort()

        if (triangle[0] == 0) {
            break
        }

        if (triangle[2] >= triangle[0] + triangle[1]) {
            println("Invalid")
        } else {
            when (triangle.groupBy { it }.size) {
                1 -> println("Equilateral")
                2 -> println("Isosceles")
                else -> println("Scalene")
            }
        }
    }
}